### PR TITLE
add developer instructions on installing and troubleshooting dep

### PR DESCRIPTION
**Pull request by mwang3-wrs**
This commit adds details on how to install the dep tool and ways
to resolve potential issues with dep during setup.

Signed-off-by: Melissa Wang <melissa.wang@windriver.com>

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -419,3 +426,79 @@ docker tag wind-river/cloud-platform-deployment-manager:debug ${MY_REGISTRY}/win
...
 docker push ${MY_REGISTRY}/wind-river/cloud-platform-deployment-manager:debug
 ```
+
+## Troubleshooting dep
+If you encounter issues with dep (eg, dep command hangs), check the following:
+
+#### ~/.cache directory
+
+Your home directory is a shared file system under /folk. Having the .cache
+directory in your folk directory can end up filling up the disk quota. Use a
```
Comment (donpenney) [inline DEVELOPER.md:436]: Remove the reference to "/folk", as this is specific to your environment.
"Your home dir may be a shared file system. If disk quotas are enabled due to limited space, the .cache directory could end up filling up your quota."

File: DEVELOPER.md [hunk 2]
```diff
@@ -419,3 +426,79 @@ docker tag wind-river/cloud-platform-deployment-manager:debug ${MY_REGISTRY}/win
...
+symbolic link instead.
+
+```bash
+mv /folk/${USER}/.cache /localdisk/loadbuild/${USER}/.cache and
```
Comment (donpenney) [inline DEVELOPER.md:440]: Replace "/folk/${USER}" references with "${HOME}"

File: DEVELOPER.md [hunk 3]
```diff
@@ -419,3 +426,79 @@ docker tag wind-river/cloud-platform-deployment-manager:debug ${MY_REGISTRY}/win
...
+ln -s /localdisk/loadbuild/${USER}/.cache /folk/${USER}/.cache
+```
+
+#### Hanging processes
+
+Ctrl-C may not completely terminate dep. Check ps -a for dep and its
```
Comment (donpenney) [inline DEVELOPER.md:446]: I think you'd want to note here that, if the "dep ensure -update" command is hung, using "ps" to look for potentially hung child processes is a great step, before killing the hung dep command.

File: DEVELOPER.md (hunk 1 above)
Comment (mwang3-wrs) [reply DEVELOPER.md]: Fixed

File: DEVELOPER.md (hunk 2 above)
Comment (mwang3-wrs) [reply DEVELOPER.md]: Fixed

File: DEVELOPER.md (hunk 3 above)
Comment (mwang3-wrs) [reply DEVELOPER.md]: Fixed

File: DEVELOPER.md [hunk 4]
```diff
@@ -125,6 +125,13 @@ sudo usermod -a -G docker ${USER}
 newgrp docker
 ```
 
+#### Dep
+Dep is a depedency management tool for Go installed to the ${GOPATH}/bin directory.
```
Comment (albailey) [inline DEVELOPER.md:129]: typo.  depedency

File: DEVELOPER.md [hunk 5]
```diff
@@ -419,3 +426,79 @@ docker tag wind-river/cloud-platform-deployment-manager:debug ${MY_REGISTRY}/win
...
+symbolic link instead.
+
+```bash
+mv ${HOME}/.cache /localdisk/loadbuild/${USER}/.cache and
+ln -s /localdisk/loadbuild/${USER}/.cache ${HOME}/.cache
+```
+
+#### Hanging processes
+
+If "dep ensure -update" command is hung, use ps -a to check for dep and its
```
Comment (albailey) [inline DEVELOPER.md:446]: nit.  You put the "dep ensure" in quotes but not the "ps -a"

**Pull request merged**